tezedge_state/src/peer_address.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) SimpleStaking, Viable Systems and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use serde::{Deserialize, Serialize};
+use std::fmt::{self, Display};
+use std::hash::Hash;
+use std::net::{AddrParseError, IpAddr, Ipv4Addr, SocketAddr};
+use std::str::FromStr;
+
+use tla_sm::recorders::CloneRecorder;
+use tla_sm::DefaultRecorder;
+
+pub type Port = u16;
+
+/// Any kind of SocketAddr.
+///
+/// Port can be listener port or some arbitrary port.
+#[derive(Serialize, Deserialize, Debug, Hash, Ord, PartialOrd, Eq, PartialEq, Clone, Copy)]
+#[repr(transparent)]
+pub struct PeerAddress(SocketAddr);
+
+impl PeerAddress {
+    pub fn new(ip: IpAddr, port: Port) -> Self {
+        Self(SocketAddr::new(ip, port))
+    }
+
+    pub fn ip(&self) -> IpAddr {
+        self.0.ip()
+    }
+
+    pub fn port(&self) -> Port {
+        self.0.port()
+    }
+
+    pub fn set_port(&mut self, new_port: Port) {
+        self.0.set_port(new_port)
+    }
+
+    /// Caller must make sure that the set port is the port that the
+    /// remote peer is listening on.
+    pub fn as_listener_address(&self) -> PeerListenerAddress {
+        PeerListenerAddress(self.0)
+    }
+
+    /// For testing purposes.
+    ///
+    /// Get unique ipv4 address based on index.
+    pub fn ipv4_from_index(index: u64) -> Self {
```

Comment:
@binier 
I am not very big fan of mixing test stuff to "pub" stuff

Reply:
I agree, we should have that under feature flag.

Reply:
It's not enough, because `simulator` crate needs it, and it's not a test module, it's more of a framework.